### PR TITLE
[MB-6834] Add pricer params for SIT First Day pricers

### PR DESCRIPTION
**Pull request by reggieriser**
## Description

This PR adds pricer params to both the domestic origin and destination pricers.

## Reviewer Notes

- I also refactored existing pricer param code to use a few new helper functions (`FormatEscalation` and `suite.validatePricerCreatedParams`) that were recently introduced by Jacquie.

## Setup

To get to the place where the DDFSIT screenshot below was taken, do the following:
- Build and start your local client/server (and make sure your DB is up-to-date)
- Create a new move in the customer UI
- As a TOO in the office UI, edit the orders to add the missing fields, then approve the move and the service items.  That should automatically add other service items like DLH.
- Run the `prime-api-demo` script locally: `prime-api-demo <move locator> local`
- Have it update dates/weights, create a payment request, and upload proof of service docs.
- Go back to the office UI as a TIO and approve the payment request and its service items.
- You should then have the ability to see the parameter details and pricing on the DDFSIT service item as shown below.
- You should also be able to see the params by looking in the `payment_service_item_params` table or running a `fetch-mto-updates` via the `prime-api-client`.

We can also add a `DOFSIT` (origin first day SIT) to the move. Use this JSON template to make a file named `create_dofsit_service_item.json`, filling in the move ID and shipment ID (which you can find in your local database from the locator):

```
{
  "body": {
    "moveTaskOrderID": "<insert MTO ID>",
    "mtoShipmentID": "<insert MTO shipment ID>",
    "modelType": "MTOServiceItemOriginSIT",
    "reServiceCode": "DOFSIT",
    "sitPostalCode": "30907",
    "reason": "my address changed",
    "sitEntryDate": "2021-01-13",
    "sitHHGActualOrigin":  {
      "streetAddress1": "20245 4th St",
      "city": "New Braunfels",
      "state": "TX",
      "postalCode": "78130",
      "country": "United States"
    }
  }
}
```

Create the `DOFSIT` service item with this command:

```
prime-api-client --insecure create-mto-service-item --filename create_dofsit_service_item.json | jq
```

Now we should be able to issue a payment request for the `DOFSIT` service item.  Create a `create_payment_request.json` file with these contents (looking up the IDs noted below in your database):

```json
{
  "body": {
    "isFinal": false,
    "moveTaskOrderID": "<insert MTO ID>",
    "serviceItems": [
      {
        "id": "<insert DOFSIT MTO service item ID>"
      }
    ]
  }
}
```

Now send this payload to create the payment request:

```
prime-api-client create-payment-request --insecure --filename create_payment_request.json | jq
```

Approve the payment request as a TIO to get to the point where you can see the DOFSIT screenshot below.

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6834) for this change

## Screenshots

Screen Shot 2021-04-13 at 1 43 18 PM

Screen Shot 2021-04-13 at 1 58 36 PM

### CODE REVIEW

File: pkg/services/ghcrateengine/counseling_services_pricer_test.go [hunk 1]
```diff
@@ -26,8 +27,10 @@ func (suite *GHCRateEngineServiceSuite) TestPriceCounselingServices() {
 		suite.Equal(csPriceCents, priceCents)
 
 		// Check that PricingDisplayParams have been set and are returned
-		suite.Equal(len(displayParams), 1)
-		suite.HasDisplayParam(displayParams, models.ServiceItemParamNamePriceRateOrFactor, FormatCents(csPriceCents))
+		expectedParams := services.PricingDisplayParams{
+			{Key: models.ServiceItemParamNamePriceRateOrFactor, Value: FormatCents(csPriceCents)},
+		}
+		suite.validatePricerCreatedParams(expectedParams, displayParams)
```
Comment (abbyoung) [inline pkg/services/ghcrateengine/counseling_services_pricer_test.go:33]: Awesome!

File: pkg/services/ghcrateengine/ghc_rate_engine_service_test.go [hunk 2]
```diff
@@ -110,10 +110,11 @@ func (suite *GHCRateEngineServiceSuite) setupDomesticOtherPrice(code models.ReSe
 	suite.MustSave(&otherPrice)
 }
 
-func (suite *GHCRateEngineServiceSuite) setupDomesticServiceAreaPrice(code models.ReServiceCode, serviceAreaCode string, isPeakPeriod bool, priceCents unit.Cents, escalationCompounded float64) {
+func (suite *GHCRateEngineServiceSuite) setupDomesticServiceAreaPrice(code models.ReServiceCode, serviceAreaCode string, isPeakPeriod bool, priceCents unit.Cents, contractYearName string, escalationCompounded float64) {
 	contractYear := testdatagen.MakeReContractYear(suite.DB(),
 		testdatagen.Assertions{
 			ReContractYear: models.ReContractYear{
+				Name:                 contractYearName,
```
Comment (abbyoung) [inline pkg/services/ghcrateengine/ghc_rate_engine_service_test.go:117]: Great, I can use this update in my PR too. Thank you!

**Review by abbyoung — APPROVED** [review]
LGTM! Thanks for making all the updates to the tests/shared functions.

**Review by pearl-truss — APPROVED** [review]
LGTM!

**Pull request merged**